### PR TITLE
DOC-8029: Docs for Enable SQL stats (Sessions, Transactions, and Statements) to be externalized with fine granularity for production workloads

### PR DESCRIPTION
**Pull request by florence-crl**
Addresses: DOC-8029

(1) Added Log SQL statistics to Datadog page with image. (2) Updated Self-Hosted Deployments sidebar, added Log SQL statistics to Datadog. (3) Updated CockroachDB Feature Availability page, Features in preview section. (4) Updated 23.2 files.

### CODE REVIEW

**Review by xinhaoz — APPROVED** [review]
LGTM, I'd also request a review from someone from obs-inf, specifically for the disabling writing to disk portion.

**Comment by florence-crl** [comment]
@dhartunian please review the "the disabling writing to disk portion."

File: src/current/v23.1/log-sql-statistics-to-datadog.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+1. Copy the newly created API key to be used in Step 2.
+
+## Step 2. Configure an HTTP network collector for Datadog
+
+Configure an [HTTP network collector] (configure-logs.html#output-to-http-network-collectors) by creating or modifying the [`log.yaml` file](configure-logs.html#yaml-payload).
+
+{{site.data.alerts.callout_danger}}
+Given the [volume of `sampled_query` events](#step-3-configure-cockroachdb-to-emit-query-events), do not write `sampled_query` events to disk, or [`file-groups`](configure-logs.html#output-to-files). Writing a high volume of `sampled_query` events to a file group will unnecessarily consume cluster resources and impact workload performance. 
+
+To avoid writing to disk and disable file creation:
```
Comment (dhartunian) [inline src/current/v23.1/log-sql-statistics-to-datadog.md:25]: I think we should reconsider this portion of the instructions. This directs customers to turn off **all** non-Telemetry logs which would be pretty dangerous for a production cluster and prevent any debugging from proceeding.

I think a better option is to provide a config that **only disables TELEMETRY** file output which is something like the following. @kevin-v-ngo can provide a suggestion here as well from the customer we built this integration for.

```
sinks:
  http-servers:
    datadog:
      channels: [TELEMETRY]
      address: https://http-intake.logs.datadoghq.com/api/v2/logs
      format: json
      method: POST
      compression: gzip
      headers: {DD-API-KEY: "{DATADOG API KEY}"} # replace with actual API key
      buffering:
        format: json-array
        max-staleness: 5s
        flush-trigger-size: 2.5MiB # override default value
        max-buffer-size: 50MiB
  file-groups:
    telemetry:  # override telemetry file defaults; leave the rest alone
      channels: []
```

**Review by dhartunian — CHANGES_REQUESTED** [review]

**Review by dhartunian — APPROVED** [review]
LGTM. Thanks!

File: src/current/v23.1/log-sql-statistics-to-datadog.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+- Override the default logging configuration for the `telemetry` file-group by setting `channels` to empty square brackets `[]`.
+{{site.data.alerts.end}}
+
+In this `logs.yaml` example:
+
+   1. To send sampled query events directly to Datadog without writing events to disk, override telemetry default configuration by setting `file-groups: telemetry: channels:` to `[]`.
+   1. To connect to Datadog, replace `{DATADOG API KEY}` with the value you copied in Step 1.
+   1. To control the ingestion and potential drop rate for sampled query events, configure the following `buffering` values depending on your workload:
+
+   - `max-staleness`: The maximum time a log message will wait in the buffer before a flush is triggered. Set to `0` to disable flushing based on elapsed time. Default: `5s`
```
Comment (gemma-shay) [inline src/current/v23.1/log-sql-statistics-to-datadog.md:42]: These could be indented for readability

File: src/current/v23.1/log-sql-statistics-to-datadog.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+~~~ sql
+SET CLUSTER SETTING sql.telemetry.query_sampling.max_event_frequency = 100000;
+~~~
+
+## Step 4. Monitor TELEMETRY logs in Datadog
+
+1. Navigate to [**Datadog > Logs**](https://app.datadoghq.com/logs).
+1. Filter by **OTHERS > channel: TELEMETRY** to see the logs for the query events that are emitted. For example:
+
+<img src="{{ 'images/v23.1/datadog-telemetry-logs.png' | relative_url }}" alt="Datadog Telemetry Logs" style="border:1px solid #eee;max-width:100%" />
```
Comment (gemma-shay) [inline src/current/v23.1/log-sql-statistics-to-datadog.md:90]: Just a suggestion, but "See also" section might be good to end the page

File: src/current/v23.1/log-sql-statistics-to-datadog.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+{{site.data.alerts.callout_danger}}
+Given the [volume of `sampled_query` events](#step-3-configure-cockroachdb-to-emit-query-events), do not write `sampled_query` events to disk, or [`file-groups`]({% link {{ page.version.version }}/configure-logs.md %}#output-to-files). Writing a high volume of `sampled_query` events to a file group will unnecessarily consume cluster resources and impact workload performance. 
+
+`sampled_query` events are a type of [telemetry event]({% link {{ page.version.version }}/eventlog.md %}#telemetry-events). The [default logging configuration]({% link {{ page.version.version }}/configure-logs.md %}#default-logging-configuration) includes a `telemetry` file-group with `channels` set to `[TELEMETRY]`.
+
+To avoid writing `sampled_query` events (and other telemetry events) to disk and disable file creation of the `telemetry` file-group:
+
+- Override the default logging configuration for the `telemetry` file-group by setting `channels` to empty square brackets `[]`.
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/v23.1/log-sql-statistics-to-datadog.md:33]: Updated to:
> To disable the creation of a telemetry file and avoid writing `sampled_query` events and other [telemetry events](http://127.0.0.1:4000/docs/v23.1/eventlog.html#telemetry-events) to disk, change the telemetry `file-groups` setting from the [default of `channels: [TELEMETRY]`](http://127.0.0.1:4000/docs/v23.1/configure-logs.html#default-logging-configuration) to `channels: []`.

File: src/current/v23.1/log-sql-statistics-to-datadog.md (hunk 2 above)
Comment (florence-crl) [reply src/current/v23.1/log-sql-statistics-to-datadog.md:42]: I was unable to indent in markdown, so I could not update.

File: src/current/v23.1/log-sql-statistics-to-datadog.md (hunk 3 above)
Comment (florence-crl) [reply src/current/v23.1/log-sql-statistics-to-datadog.md:90]: Added "See also" section.

**Review by gemma-shay — APPROVED** [review]
looks good!

**Review by kevin-v-ngo — COMMENTED** [review]
Looks good Florence. Just one feedback on guidance on the cluster setting and buffering options.

File: src/current/v23.1/log-sql-statistics-to-datadog.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+## Step 3. Configure CockroachDB to emit query events
+
+Enable the [`sql.telemetry.query_sampling.enabled` cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-sql-telemetry-query-sampling-enabled) so that executed queries will emit an event on the telemetry [logging channel]({% link {{ page.version.version }}/logging-overview.md %}#logging-channels):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING sql.telemetry.query_sampling.enabled = true;
+~~~
+
+Set the `sql.telemetry.query_sampling.max_event_frequency` cluster setting to `100000` to emit query events at a higher rate per second than the default value of `8`, which is extremely conservative for the Datadog HTTP API. This cluster setting controls the max event frequency at which CockroachDB samples queries for telemetry.
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/v23.1/log-sql-statistics-to-datadog.md:74]: updated

**Review by kevin-v-ngo — APPROVED** [review]
Thanks for the update!

**Pull request merged**